src/Library/public/usbguard/DeviceManager.hpp

```diff
@@ -44,23 +44,38 @@ namespace usbguard
       Remove = 3,
     };
 
+    enum class AuthorizedDefaultType {
```

Comment:
According to the [kernel documentation](https://www.kernel.org/doc/html/latest/usb/authorization.html) about usb authorization there are only 3 different values the `authorized_default` file can hold, including:
-  "0" New devices connected to hostX to be deauthorized by default
-  "1" New devices connected to hostX to be authorized by default
-  "2" This causes kernel to only authorize by default devices connected to internal USB ports.

However, when AuthorizedDefault is set to 'wired' in the usbguard-daemon.conf.in file, the daemon will attempt to write a "-1" to the appropriate authorized_default file, which will fail.

Reply:
I am checking to see if this is due to a change in the patch that landed in the CrOS kernel and the one that landed in upstream. If so, I will work to resolve it.

Reply:
It looks like it is a documentation problem. See the source code here:
https://github.com/torvalds/linux/blob/master/drivers/usb/core/hcd.c#L369

Reply:
Could you please verify that AuthorizedDefault=wired works for you?  Thank you in advance .

Reply:
It works for me, but it may not work for people running an older kernel versions which is fine as long as usbguard-daemon falls back to 0.

Reply:
May I ask which version of the kernel(and linux distribution) you are using?

Reply:
I took a look at this again and I was confusing wired(-1) with internal(2). I had done some work earlier to get internal devices added.

My understanding is -1 was present because of wireless USB, but I don't know why setting -1 doesn't work for a typical hub. I am hesitant to remove it without understanding why it is still present in the kernel sources.

Reply:
I agree with you. This particular part of code in the kernel is confusing. According to the source documentation it should be doable, however it's not.

I've tried it on older kernel versions to verify that this is a kernel bug but I had no luck making it to work.

Reply:
After some digging in kernel sources it seems that `wired` was never an option for setting in `authorized_default`. It was implemented as a kernel internal state for enabling wired controllers during boot. The USBGuard daemon is not running at this point so kernel boots up with a limited set of devices. Setting `authorized_default` to `wired` would be useless after boot because only host controller can handle the value and it is already allowed.

Can anybody confirm my findings?

Reply:
@Allen-Webb @Cropi I believe we should remove the `wired` option totally from the USBGuard.

Reply:
Yes, I agree with you.